paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/CustomerSheetTest.kt

```diff
@@ -215,83 +246,99 @@ internal class CustomerSheetTest {
             response.testBodyFromFile("payment-methods-create.json")
         }
 
-        networkRule.enqueue(
-            retrieveSetupIntentRequest(),
-            retrieveSetupIntentParams(),
-        ) { response ->
-            response.testBodyFromFile("setup-intent-get.json")
-        }
-
-        networkRule.enqueue(
-            confirmSetupIntentRequest(),
-            confirmSetupIntentParams(),
-        ) { response ->
-            response.testBodyFromFile("setup-intent-confirm.json")
+        when (customerSheetTestType) {
+            CustomerSheetTestType.AttachToCustomer -> {
+                networkRule.enqueue(
+                    attachPaymentMethodRequest(),
+                ) { response ->
+                    response.testBodyFromFile("payment-methods-create.json")
+                }
+            }
+            CustomerSheetTestType.AttachToSetupIntent -> {
+                networkRule.enqueue(
+                    retrieveSetupIntentRequest(),
+                    retrieveSetupIntentParams(),
+                ) { response ->
+                    response.testBodyFromFile("setup-intent-get.json")
+                }
+
+                networkRule.enqueue(
+                    confirmSetupIntentRequest(),
+                    confirmSetupIntentParams()
+                ) { response ->
+                    response.testBodyFromFile("setup-intent-confirm.json")
+                }
+            }
         }
 
         page.clickSaveButton()
         page.clickConfirmButton()
     }
 
     @Test
-    fun testCardNotSavedOnConfirmError() = activityScenarioRule.runCustomerSheetTest(
-        integrationType = integrationType,
-        resultCallback = {
-            error("Shouldn't call CustomerSheetResultCallback")
-        }
-    ) { context ->
-        networkRule.enqueue(
-            retrieveElementsSessionRequest(),
-        ) { response ->
-            response.testBodyFromFile("elements-sessions-requires_payment_method.json")
-        }
-
-        networkRule.enqueue(
-            retrievePaymentMethodsRequest(),
-            cardPaymentMethodsParams(),
-        ) { response ->
-            response.testBodyFromFile("payment-methods-get-success-empty.json")
-        }
-
-        networkRule.enqueue(
-            retrievePaymentMethodsRequest(),
-            usBankAccountPaymentMethodsParams(),
-        ) { response ->
-            response.testBodyFromFile("payment-methods-get-success-empty.json")
-        }
-
-        context.presentCustomerSheet()
-
-        page.fillOutCardDetails()
-
-        networkRule.enqueue(
-            createPaymentMethodsRequest(),
-            cardDetailsParams(),
-            billingDetailsParams(),
-        ) { response ->
-            response.testBodyFromFile("payment-methods-create.json")
+    fun testCardNotAttachedOnError() {
+        assume().that(customerSheetTestType).isNotEqualTo(CustomerSheetTestType.AttachToCustomer)
```

Comment:
Skips this test on the `AttachToCustomer` case. Will be removing it in a follow up PR with an error case for attaching a card to a customer.

Reply:
We should be able to avoid this by adding `customerSheetTestType` as a parameter to the function, rather than the class. See https://github.com/stripe/stripe-android/blob/ebba90908de294da173307ebe92d2be85a76bb5e/financial-connections/src/test/java/com/stripe/android/financialconnections/screenshottests/FinancialConnectionsShotShowkaseScreenshotTest.kt#L59

For this test, we wouldn't add it, but for the ones we want to test both scenarios we would add it.